### PR TITLE
Add spill support for oscar

### PR DESCRIPTION
**Pull request by hekaisheng**
## What do these changes do?

Add spill support for storage service.

### CODE REVIEW

**Review by sighingnow — APPROVED** [review]
LGTM.

File: mars/storage/plasma.py [hunk 1]
```diff
@@ -210,6 +210,9 @@ async def get(self, object_id, **kwargs) -> object:
         if kwargs:  # pragma: no cover
             raise NotImplementedError('Got unsupported args: {",".join(kwargs)}')
 
+        if object_id not in self._client.list():  # pragma: no cover
```
Comment (qinxuye) [inline mars/storage/plasma.py:213]: Use `contains` instead of `list`.

File: mars/storage/plasma.py [hunk 2]
```diff
@@ -256,6 +259,8 @@ async def open_writer(self, size=None) -> StorageFileObject:
 
     @implements(StorageBackend.open_reader)
     async def open_reader(self, object_id) -> StorageFileObject:
+        if object_id not in self._client.list():  # pragma: no cover
```
Comment (qinxuye) [inline mars/storage/plasma.py:262]: Ditto.

**Review by qinxuye — COMMENTED** [review]
LGTM overall, some comments are left.

**Review by qinxuye — APPROVED** [review]
LGTM

File: mars/services/storage/core.py [hunk 3]
```diff
@@ -184,39 +232,117 @@ def get_info(self,
...
+                     f'in data manager: {to_delete_keys}')
+
+    @extensible
+    def delete_data_info(self,
+                         session_id: str,
+                         data_key: str,
+                         level: StorageLevel):
+        self._delete_data_info(session_id, data_key, level)
+
+    @delete_data_info.batch
```
Comment (wjsi) [inline mars/services/storage/core.py:306]: Batch method not needed for sync methods.

File: mars/services/storage/core.py [hunk 4]
```diff
@@ -184,39 +232,117 @@ def get_info(self,
...
+    @extensible
+    async def put_data_info(self,
+                            session_id: str,
+                            data_key: str,
+                            data_info: DataInfo,
+                            object_info: ObjectInfo = None):
+        await self._put_data_info(
+            session_id, data_key, data_info, object_info=object_info)
+
+    @put_data_info.batch
```
Comment (wjsi) [inline mars/services/storage/core.py:270]: Batch method not needed for sync methods.

File: mars/services/storage/core.py [hunk 5]
```diff
@@ -184,39 +232,117 @@ def get_info(self,
...
-               data_key: str,
-               level: StorageLevel):
+    @extensible
+    def get_data_info(self,
+                      session_id: str,
+                      data_key: str,
+                      error: str = 'raise') -> Union[DataInfo, None]:
+        return self._get_data_info(session_id, data_key, error)
+
+    @get_data_info.batch
```
Comment (wjsi) [inline mars/services/storage/core.py:242]: Batch method not needed for sync methods.

File: mars/services/storage/core.py [hunk 6]
```diff
@@ -152,27 +176,51 @@ def put(self,
...
+                return
+
+    @extensible
+    def get_data_infos(self,
+                       session_id: str,
+                       data_key: str,
+                       error: str = 'raise') -> List[DataInfo]:
+        return self._get_data_infos(session_id, data_key, error)
+
+    @get_data_infos.batch
```
Comment (wjsi) [inline mars/services/storage/core.py:213]: Batch method not needed for sync methods.

Comment (hekaisheng) [reply mars/services/storage/core.py]: Removed

**Review by wjsi — APPROVED** [review]
LGTM

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**